internal/cmdget/cmdget.go

```diff
@@ -87,20 +87,15 @@ func (r *Runner) preRunE(_ *cobra.Command, args []string) error {
 			args[1] = resolvedPath
 		}
 	}
-	destination, err := parse.ParseArgs(r.ctx, args, parse.Options{
-		SetGit: func(git *kptfilev1.Git) error {
-			r.Get.Git = git
-			r.Get.Upstream = remote.NewGitUpstream(git)
-			return nil
-		},
-		SetOci: func(oci *kptfilev1.Oci) error {
-			r.Get.Upstream = remote.NewOciUpstream(oci)
-			return nil
-		},
-	})
+
+	upstream, destination, err := parseref.ParseArgs(
+		args,
+		location.WithContext(r.ctx),
+		location.WithParsers(location.GitParser, location.OciParser))
 	if err != nil {
-		return err
+		return errors.E(op, err)
```

Comment:
so one change I see here is that new code is wrapping the error. It may potentially impact the error message that gets displayed on the CLI.